packages/core/components/EditorPanel/DataTableEditor.tsx

```diff
@@ -75,7 +75,7 @@ const DataTable: React.FC<DataTableProps> = ({ config, updateField, isDashboard,
           }
         />
       )}
-      <TextField value={config.table.indexLabel} section='table' fieldName='indexLabel' label='Index Column Header' updateField={updateField} />
+      <TextField value={config.table.indexLabel ? config.table.indexLabel : config.xAxis?.dataKey} section='table' fieldName='indexLabel' label='Index Column Header' updateField={updateField} />

```

Comment:
I think what we need here is the empty string still. Then, in the table header output we can check for the empty string and if it is empty we can output the content that's mentioned in the ticket `<th><span sr-only>Not Applicable</span></th>`